addons/web/static/tests/views/form/form_view_tests.js

```diff
@@ -12803,4 +12803,26 @@ QUnit.module("Views", (hooks) => {
         await click(target.querySelector(".o_data_row .o_data_cell"));
         assert.containsOnce(target, ".o_form_view");
     });
+
+     QUnit.test("archive action not shown with readonly active field", async function (assert) {
+
+        // add active field on partner model, but do not put it in the view
+        serverData.models.partner.fields.active = {
+            string: "Active",
+            type: "char",
+            default: true,
+            readonly: true,
+        };
+        await makeView({
+            type: "form",
+            resModel: "partner",
+            resId: 1,
+            serverData,
+            arch: `<form><field name="foo"/></form>`,
+            actionMenus: {},
+        });
+        await click(target.querySelector(".o_cp_action_menus"))
+        assert.containsNone(target, ".o_menu_item:contains(Archive)");
```

Comment:
try something like this:
```suggestion
        await click(target, ".o_cp_action_menus");
        assert.deepEqual(
            [...target.querySelectorAll(".o_menu_item")].map((el) => el.textContent),
            ["action 1", "action 2"], // ...
            "should not contain an Archive action"
        );
```

Reply:
when you debug the test, inspect the dropdown and see how many menu items you should see.

Maybe "o_menu_item" is not the proper dom selector.

Reply:
You should see this
![image](https://user-images.githubusercontent.com/1159815/211821653-7269dfd5-9118-4342-badf-793d4747350e.png)

Reply:
Exactly

Reply:
replace selector
```
.o_cp_action_menus
``` 

with
```
.o_cp_action_menus .dropdown-toggle
```